lib/rspec/core/formatters/html_formatter.rb

```diff
@@ -69,9 +69,11 @@ def example_failed(failure)
           example = failure.example
 
           exception = failure.exception
+          message_lines = failure.fully_formatted_lines(nil, RSpec::Core::Notifications::NullColorizer)
           exception_details = if exception
                                 {
-                                  :message => failure.message_lines.join("\n"),
+                                  # the regular expression here removes the CLI leading indentation, and any trailing spaces.
+                                  :message => message_lines.drop(2).join("\n").gsub(/^  (.*?)( +)?$/, '\1'),
```

Comment:
The regex makes me a little nervous...

* I'm having a hard time grokking it, which isn't good for long-term maintainability.  In general, I like to leave a comment with one or more rubular permalinks (such as [here](https://github.com/rspec/rspec-core/blob/56e31e3a5cbeb296082e71e2fb46b6c98ef798a5/lib/rspec/core/example.rb#L123-L124)) -- that way, our future selves can click through to see examples of text the regex did and did not match against.
* I think part of the complexity of the regex is due to the fact that it is intended to affect both the start and the end of the string (for the trailing spaces). I suspect it would be simpler to make the regex do only the start of the string, and use `String#rstrip` to strip trailing spaces (since that's the entire purpose of `rstrip`....).
* The regex itself seems kinda brittle.  For example, the start (the `/^  ` bit) seems to assume two spaces of indentation -- but we don't know if that assumption will always hold true.  The goal is to unindent, whatever the level of indentation is, right?  For that, I think we're better of using the [unindent helper](https://github.com/rspec/rspec-core/blob/56e31e3a5cbeb296082e71e2fb46b6c98ef798a5/spec/support/aruba_support.rb#L52-L57) I snagged from Ruby tapas awhile back.  Basically, it works on a string with any level of indentation. It works by finding the smallest leading whitespace of any line, treating that as the indentation level of the whole string, and `gsub`ing it out on every line.  Really simple and robust.